### PR TITLE
SUBMARINE-388. Rebuild submarine image every run k8s-test

### PR DESCRIPTION
**Pull request by liuxunorg**
### What is this PR for?

k8s test: When the user submits the code to his/her repository or the `apache/submarine` git repository, the travis test task will automatically start.

test-k8s runs test cases in travis. It will first create a k8s cluster by using the kind tool in travis,

and then compile and package the submarine project in `submarine-dist` directory to build a docker image.

Then use this latest code build docker image in k8s Deploy a submarine system. Then run test case in the `test-k8s/..` directory.

### What type of PR is it?
[Feature]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/SUBMARINE-388

### How should this be tested?
* https://travis-ci.org/liuxunorg/submarine/builds/650831518

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes

### CODE REVIEW

File: docs/development/IntegrationTest.md [hunk 1]
```diff
@@ -1,31 +1,56 @@
...
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
 -->
 
 # IntegrationTest
 
-## Run the existing tests.
+Submarine now supports two integration tests.
```
[the remark this answers is not in the document]
Comment (liuxunorg) [reply docs/development/IntegrationTest.md]: Done.

File: docs/development/IntegrationTest.md [hunk 2]
```diff
@@ -1,31 +1,56 @@
...
+
+They are in the project's `submarine/submarine-test` directory, There are two modules, `e2e` and `test-k8s`.
+
+There are currently some differences between `e2e` and `test-k8s` in operation mode. Among them, `e2e` needs to deploy submarine locally, while `test-k8s` uses k8s to deploy submarine.
+
+These two different test methods can be applied to different test scenarios. (In the future, these two test methods may be combined or adjusted)
+
+## k8s test
+
+k8s test: When the user submits the code to himself or the `apache/submarine` git repository, the travis test task will automatically start.
```
[the remark this answers is not in the document]
Comment (liuxunorg) [reply docs/development/IntegrationTest.md]: Done.

File: docs/development/IntegrationTest.md [hunk 3]
```diff
@@ -1,31 +1,56 @@
...
+
+test-k8s runs test cases in travis. It will first create a k8s cluster by using the kind tool in travis, and then compile and package the submarine project in `submarine-dist` directory to build a docker image.
+
+Then use this latest code build docker image in k8s Deploy a submarine system. Then run test case in the `test-k8s/test` directory.
+
+
+## e2e test
+
+e2e tests can be unit tested locally and in travis,
```
[the remark this answers is not in the document]
Comment (liuxunorg) [reply docs/development/IntegrationTest.md]: Done.

File: docs/development/IntegrationTest.md [hunk 4]
```diff
@@ -1,31 +1,79 @@
...
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
 -->
 
 # IntegrationTest
 
-## Run the existing tests.
+Submarine now supports two kind of integration tests.
```
Comment (yuanzac) [inline docs/development/IntegrationTest.md]: Change kind to kinds

File: docs/development/IntegrationTest.md [hunk 5]
```diff
@@ -1,31 +1,79 @@
...
+
+## k8s test
+
+k8s test: When the user submits the code to his/her repository or the `apache/submarine` git repository, the travis test task will automatically start.
+
+test-k8s runs test cases in travis. It will first create a k8s cluster by using the kind tool in travis,
+
+and then compile and package the submarine project in `submarine-dist` directory to build a docker image.
+
+Then use this latest code build docker image in k8s Deploy a submarine system. Then run test case in the `test-k8s/..` directory.
```
Comment (yuanzac) [inline docs/development/IntegrationTest.md]: Then use this latest code to build a docker image and deploy a submarine system in k8s.

File: submarine-cloud/Makefile [hunk 6]
```diff
@@ -34,20 +33,19 @@ LDFLAGS		:= -s -w \
 
 .PHONY: build
 build:
-	$(GOBUILD) -o ./bin/$(BINARY_NAME) -v cmd/operator/main.go
+	GOOS=${GOOS} GOARCH=amd64 $(GOBUILD) -o ./bin/$(BINARY_NAME) -v cmd/operator/main.go
 
 test:
-	$(GOTEST) -v ./...
+	#$(GOTEST) -v ./...
```
Comment (yuanzac) [inline submarine-cloud/Makefile]: Delete this line instead of add a comment "#"?

Comment (liuxunorg) [reply submarine-cloud/Makefile]: Now, Test cases for submarine-cloud are not yet supported.
I will support it in next work.

File: docs/development/IntegrationTest.md (hunk 5 above)
Comment (liuxunorg) [reply docs/development/IntegrationTest.md]: Done.

File: docs/development/IntegrationTest.md (hunk 4 above)
Comment (liuxunorg) [reply docs/development/IntegrationTest.md]: Done.

**Comment by liuxunorg** [comment]
@tangzhankun @yuanzac  Thank you help review this PR.
Will merge if no more comments.

**Pull request closed without merging**